server/webapp/WEB-INF/rails.new/config/routes.rb

```diff
@@ -19,7 +19,7 @@
   mount JasmineRails::Engine => '/jasmine-specs-new', as: :jasmine_new if defined?(JasmineRails)
 
   unless defined?(CONSTANTS)
-    USER_NAME_FORMAT = GROUP_NAME_FORMAT = TEMPLATE_NAME_FORMAT = PIPELINE_NAME_FORMAT = STAGE_NAME_FORMAT = ENVIRONMENT_NAME_FORMAT = /[\w\-][\w\-.]*/
+    USER_NAME_FORMAT = GROUP_NAME_FORMAT = TEMPLATE_NAME_FORMAT = PIPELINE_NAME_FORMAT = STAGE_NAME_FORMAT = ENVIRONMENT_NAME_FORMAT = PLUGIN_ID_FORMAT = /[\w\-][\w\-.]*/
```

Comment:
Plugin IDs have this format — https://github.com/gocd/gocd/blob/242d113bb7724911ee0c7166b200d56a86f3ada3/plugin-infra/go-plugin-api/resources/plugin-descriptor.xsd#L30-L37